lua/neotest-kotlin/src/output-parser.lua

```diff
@@ -1,140 +1,110 @@
--- Example Output: '/home/cody/dev/src/git-down/src/test/kotlin/com/codymikol/state/neotest.kt::"NeotestKotestSpec"::"a namespace"::"should handle passed assertions"'
-M.make_result_id = function(line, path)
-  --
-  -- get the line starting after the PREFIX_OFFSET
-  local line_without_prefix = string.sub(line, PREFIX_OFFSET + 2)
-
-  local id = path
-
-  local parts = vim.split(line_without_prefix, " > ", { trimempty = true })
-
-  -- Cleaning the fully qualified class name
-
-  local fullyQualifiedClassName = parts[1]
-
-  local fullyQualifiedClassNameParts = vim.split(fullyQualifiedClassName, "%.")
-
-  local className = fullyQualifiedClassNameParts[#fullyQualifiedClassNameParts]
-
-  parts[1] = className
-
-  table.remove(parts, 1)
-
-  -- cleaning the it description that is for some reason prepended with the fully qualified class name...
-
-  local itDescription = parts[#parts]
-
-  local cleanedItDescription = string.sub(itDescription, fullyQualifiedClassName:len() + 2)
-
-  parts[#parts] = cleanedItDescription
-
-  for _, value in ipairs(parts) do
-    id = id .. "::" .. '"' .. value .. '"'
-  end
-
-  -- I'm sure there is a better way to do this, but I'm sleepy...
-
-  if string.find(id, ' PASSED"', -8) then
-    return string.sub(id, 1, -9) .. '"'
-  end
-
-  if string.find(id, ' SKIPPED"', -9) then
-    return string.sub(id, 1, -10) .. '"'
-  end
-
-  if string.find(id, ' FAILED"', -8) then
-    return string.sub(id, 1, -9) .. '"'
-  end
-
-  return result
+-- org.example.KotestDescribeSpec > a namespace > should handle failed assertions FAILED
+-- '/home/nick/GitHub/neotest-kotlin/lua/tests/example_project/app/src/test/kotlin/org/example/KotestDescribeExample.kt::"a namespace"::"a nested namespace"::"should handle failed assertions"'
+---Parses the Neotest id from a Gradle test line output
+---@param line string
+---@param path string
+---@param package string
+---@return string? neotest_id
+M.parse_test_id = function(line, path, package)
+	if not M.is_valid_gradle_test_line(line, package) then
+		return nil
+	end
+
+	local split = vim.split(line, ">", { trimempty = true })
+	-- Must have at least "fqn > test"
+	if #split < 2 then
+		return nil
+	end
+
+	local names = { unpack(split, 2) }
+
+	local result = path
+	for i, segment in ipairs(names) do
+		segment = vim.trim(segment)
+		if (i + 1) == #split then
+			segment = segment:match("(.+) [PASSED|FAILED|SKIPPED]")
+		end
+
+		if vim.startswith(segment, package .. ".") then
+			segment = segment:sub(#package + 2)
+		end
+
+		result = result .. '::"' .. segment .. '"'
+	end
+
+	return result
 end
 
-local function escape_magic_chars(str)
-  return str:gsub("([%^%$%(%)%%%.%[%]%*%+%-%?])", "%%%1")
-end
+---Whether the line is a valid gradle test line
+---@param line string
+---@return boolean
+function M.is_valid_gradle_test_line(line, package)
```

Comment:
Simple helper here to ensure that the line is one that we can parse and care about